core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java

```diff
@@ -295,9 +296,11 @@ public void write(String name, String value) throws IOException {
     }
 
     private void writeName(String name) throws IOException {
+        String truncatedString = truncate(name, 150);
+        String sanitizedName = SanitizationUtils.sanitizeStringForJSON(truncatedString, true);
```

Comment:
I'd recommend to look at existing libraries for json processing, such as [Jackson](https://github.com/FasterXML/jackson) where you can use `ObjectMapper` to write string/object as json.